src/auto-trace.js

```diff
@@ -1,6 +1,32 @@
 import { wrapObjectWithError, createError } from './auto-trace.helper.js';
 let globalMiddlewares = [];
 
+/**
+ * Returns a method that will handle error processing and throwing. Pass this as your onError arg in RxJS subscriptions.
+ * This is meant to replace all the other catch and stacktrace methods. https://xkcd.com/927/
+ *
+ * RxJS Subscription example:
+ *   myObs.subscribe(onComplete, catchError())
+ *
+ * If a callback is passed then that will be called with the processed error instead of throwing the error.
+ * e.g.
+ *   myObs.subscribe(onComplete, catchError((error, throwError) => {
+ *     error.showToast = false
+ *     throwError(error)
+ *   }))
+ */
+export function catchError(callback, extraContext) {
+  return asyncStacktrace((err) => {
+    if (callback) {
+      callback(err, function throwErr(err) {
+        setTimeout(() => { throw err })
+      }, extraContext)
```

Comment:
`extraContext` doesn't need to be passed to the callback does it?

Reply:
it doesn't have to, it's more for convenience of access if you just throw in an object as extraContext
```
    myObs.subscribe(onComplete, catchError((error, throwError, context) => {
      error.showToast = false
      infoToast(context.someStuff)
      throwError(error)
    }, {
      someStuff: 1
    }))
```

Reply:
is there a reason to throw inside a setTimeout? I think this will force the throw to happen outside of any micro task (like a promise) and into the main event loop...